engine/runtime/src/main/java/org/enso/interpreter/node/callable/function/BlockNode.java

```diff
@@ -89,10 +82,20 @@ private static boolean isNodeWrapped(ExpressionNode node) {
     return node instanceof StatementNode || ExpressionNode.isWrapper(node);
   }
 
+  @Override
+  public SourceSection getSourceSection() {
+    var ss = super.getSourceSection();
+    return ss != null ? ss : getRootNode().getSourceSection();
+  }
+
   @Override
   public boolean hasTag(Class<? extends Tag> tag) {
-    return super.hasTag(tag)
-        || tag == StandardTags.RootBodyTag.class
-        || tag == StandardTags.RootTag.class;
+    if (super.hasTag(tag)) {
+      return true;
+    }
+    if (tag == StandardTags.RootBodyTag.class || tag == StandardTags.RootTag.class) {
+      return true;
+    }
+    return false;
```

Comment:
One reads first `if`, then the next and then the last statement. I'd say it is a _separation of concerns_.

Reply:
One is a one-linear, the other is 7 lines. I _guess_ it could be easier for the debugger's breakpoints. But then I don't write code for the debugger but for the reader.